### PR TITLE
Add workflows

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
- add workflows for ubuntu1604, ubuntu1804, windows16, windows19
- add condition to run on label

### CODE REVIEW

File: .github/workflows/ubuntu1604.yml [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
 on:
   pull_request:
-    paths: "images/linux/**"
     types: [labeled]
 
 jobs:
   build:
-    name: ubuntu1604 job
-    runs-on: ubuntu-latest
+    name: Build ubuntu 1604
```
Comment (maxim-lobanov) [inline .github/workflows/ubuntu1604.yml]: Could you please change it to the consistent title `Ubuntu 1604`. The same for other workflows

File: .github/workflows/ubuntu1604.yml [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
+    runs-on: self-hosted
     if: github.event.label.name == 'test-ubuntu1604'
 
     steps:
-      - name: Build ubuntu1604
+      - name: Set build number
         run: |
-          echo "Long build for ubuntu1604"
\ No newline at end of file
+            $ResourcesNamePrefix = date +%s
```
Comment (maxim-lobanov) [inline .github/workflows/ubuntu1604.yml]: Let's rename this variable to `UniqueBuildNumber`

File: .github/workflows/ubuntu1604.yml [hunk 3]
```diff
@@ -1,16 +1,45 @@
...
+                            -ResourceGroup ${{ secrets.AZURE_RESOURCE_GROUP }} `
+                            -StorageAccount ${{ secrets.AZURE_STORAGE_ACCOUNT }} `
+                            -SubscriptionId ${{ secrets.AZURE_SUBSCRIPTION }} `
+                            -TenantId ${{ secrets.AZURE_TENANT }} `
+                            -Location ${{ secrets.AZURE_LOCATION }} `
+                            -VirtualNetworkName ${{ secrets.BUILD_AGENT_VNET_NAME }} `
+                            -VirtualNetworkRG ${{ secrets.BUILD_AGENT_VNET_RESOURCE_GROUP }} `
+                            -VirtualNetworkSubnet ${{ secrets.BUILD_AGENT_SUBNET_NAME }} `
+                            -GitHubFeedToken ${{ secrets.GITHUB_TOKEN }}
+        shell: pwsh
```
Comment (maxim-lobanov) [inline .github/workflows/ubuntu1604.yml]: Let's put it between `working-directory` and `run`

File: .github/workflows/ubuntu1604.yml (hunk 1 above)
Comment (dmitry-shibanov) [reply .github/workflows/ubuntu1604.yml]: Done

**Pull request merged**